URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h

```diff
@@ -31,30 +31,148 @@
+ @param statusCode The status code for the HTTP response.
+ @param responseXML A XML string to use as the mock HTTP response body.
+ @result The newly created mock HTTP GET request. Its responder is set to the new mock XML body responder.
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPGetRequestWithURL:(NSURL *)URL responseStatusCode:(NSInteger)statusCode responseXML:(NSString *)responseXML;
+
+/*!
+ @abstract Creates and expects a new mock HTTP PATCH request that responds with a XML string.
+ @discussion There is no need to add the returned mock request to the list of expected mock requests, as
+ this method adds it automatically.
+ @param URL The URL for the mock PATCH request. May not be nil.
+ @param requestXML A XML string to use as the mock HTTP PATCH request body.
+ @param statusCode The status code for the HTTP response.
+ @param responseXML A XML string to use as the mock HTTP response body.
+ @result The newly created mock HTTP PATCH request. Its responder is set to the new mock XML body responder.
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPPatchRequestWithURL:(NSURL *)URL requestXML:(NSString *)requestXML
+                                       responseStatusCode:(NSInteger)statusCode responseXML:(NSString *)responseXML;
+
+/*!
+ @abstract Creates and expects a new mock HTTP POST request that responds with a XML string.
+ @discussion There is no need to add the returned mock request to the list of expected mock requests, as
+ this method adds it automatically.
+ @param URL The URL for the mock POST request. May not be nil.
+ @param requestXML A XML string to use as the mock HTTP POST request body.
+ @param statusCode The status code for the HTTP response.
+ @param responseXML A XML string to use as the mock HTTP response body.
+ @result The newly created mock HTTP POST request. Its responder is set to the new mock XML body responder.
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPPostRequestWithURL:(NSURL *)URL requestXML:(NSString *)requestXML
+                                      responseStatusCode:(NSInteger)statusCode responseXML:(NSString *)responseXML;
+
+/*!
+ @abstract Creates and expects a new mock HTTP PUT request that responds with a XML string.
+ @discussion There is no need to add the returned mock request to the list of expected mock requests, as
+ this method adds it automatically.
+ @param URL The URL for the mock PUT request. May not be nil.
+ @param requestXML A XML string to use as the mock HTTP PUT request body.
+ @param statusCode The status code for the HTTP response.
+ @param responseXML A XML string to use as the mock HTTP response body.
+ @result The newly created mock HTTP PUT request. Its responder is set to the new mock XML body responder.
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPPutRequestWithURL:(NSURL *)URL requestXML:(NSString *)requestXML
+                                     responseStatusCode:(NSInteger)statusCode responseXML:(NSString *)responseXML;
```

Comment:
Move all these methods below the JSON ones.

Also, where you‘ve been saying requestXML: and responseXML:, rename those to requestXMLString: and responseXMLString: